### ISSUE TITLE
Create Applications with complete schema

### ISSUE THREAD

**Issue by exitcode0** [okta/terraform-provider-okta#1805]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Many `preconfigured_app` (OIN) apps are created with a subset of their possible application schema, enabling provisioning can extend an app's schema to include additional attributes.

This is problematic as it makes re-using Okta Terraform provider config difficult, Using existing configuration to provision a new application instance can result in `Apply` failures as expected schema attributes will be unavailable

I propose that the Create Application API adds a parameter that ensures that applications are created with a complete schema
This small change would allow for easier configuration re-use and works around the lack of a provisioning settings API

### New or Affected Resource(s)

- okta_app_saml
- okta_app_oauth
- okta_app_bookmark
- okta_app_swa

### Potential Terraform Configuration

```hcl
resource "okta_app_saml" "figma" {
  status            = "ACTIVE"
  label             = "Figma"
  preconfigured_app = "figma"
  app_links_json    = jsonencode({ figma_link = true })
  app_settings_json = jsonencode({
    nameIdValue = ""
    tenantId    = ""
  })
  user_name_template      = "$${source.login}"
  user_name_template_type = "BUILT_IN"

  full_schema = true
}
```

### References

- #24
- #282
- #312
- #636
- #883
- #1022
- #1409
-

**Comment by jefftaylor-okta** [comment]
Hey @exitcode0! This is definitely high on our list of enhancements. We are working with the internal teams to create a more predictable schema for all 7,500 OIN Applications. Naturally, this will take some time, but look for more OIN apps documented next year!

**Comment by exitcode0** [comment]
Hey @jefftaylor-okta 
It's great to hear that this is something that's already being worked on!
Is this something that's rolling out to each application as and when the required changes have been made, or is this being bundled up into a beta, EA, GA release cycle?

**Comment by jefftaylor-okta** [comment]
Yes, we will adding the Terraform resources as the API from the internal teams go EA. EA for us means the API contract is solid.

**Comment by duytiennguyen-okta** [comment]
OKTA internal reference https://oktainc.atlassian.net/browse/OKTA-669851
